File: solutions/kotlin/bob/1/src/main/kotlin/Bob.kt
object Bob {

    fun isQuestion(input: String): Boolean {
        return input.endsWith('?')
    }

    fun isYelling(input: String): Boolean {
        return input.uppercase() == input  && input.matches(Regex("^.*[A-Z]+.*\$"))

    }

    fun hey(input: String): String {
        val cleaned = input.trim()
        return when {
            isQuestion(cleaned) && isYelling(cleaned) -> "Calm down, I know what I'm doing!"
            isQuestion(cleaned) -> "Sure."
            isYelling(cleaned) -> "Whoa, chill out!"
            cleaned.isEmpty() -> "Fine. Be that way!"
            else -> "Whatever."
        }
    }
}

File: solutions/go/bob/1/bob.go
package bob

import (
	"regexp"
	"strings"
)

func isQuestion(remark string) bool {
	return strings.HasSuffix(remark, "?")
}

func isYelling(remark string) bool {
	regex := regexp.MustCompile(`^.*[A-Z]+.*$`)
	return strings.ToUpper(remark) == remark  && regex.MatchString(remark)
}


// Hey determines Bob's answer
func Hey(remark string) string {
	cleanedRemark := strings.TrimSpace(remark)
	switch {
	case isQuestion(cleanedRemark) && isYelling(cleanedRemark):
		return "Calm down, I know what I'm doing!"
	case isQuestion(cleanedRemark):
		return "Sure."
	case isYelling(cleanedRemark):
		return "Whoa, chill out!"
	case cleanedRemark == "":
		return "Fine. Be that way!"
	default:
		return "Whatever."
	}
}
